packages/SwingSet/src/controller.js

```diff
@@ -185,38 +185,17 @@ function makeSESEvaluator() {
     '@agoric/nat': Nat,
   });
 
-  // Augment the global meter with one that integrates refilling.
-  const refillers = new Set();
-  const refillingReplaceGlobalMeter = (meter, refillFn = undefined) => {
-    let ret;
-    if (meter === null) {
-      // Install the null meter right away.
-      ret = replaceGlobalMeter(meter);
-    }
-    if (refillFn === true) {
-      // Magic value to indicate the end of the turn.
-      refillers.forEach(refiller => refiller());
-      refillers.clear();
-    } else if (refillFn) {
-      // Just add us to the list for later refilling.
-      refillers.add(refillFn);
-    }
-    if (meter !== null) {
-      // Install the non-null meter as late as possible.
-      ret = replaceGlobalMeter(meter);
-    }
-    return ret;
-  };
-
   return src => {
     // FIXME: Note that this replaceGlobalMeter endowment is not any
     // worse than before metering existed.  However, it probably is
     // only necessary to be added to the kernel, rather than all
     // static vats once we add metering support to the dynamic vat
     // implementation.
+    // Same for registerEndOfCrank.
```

Comment:
`registerEndOfCrank` is a primal-Realm function, whereas `replaceGlobalMeter` was evaluated in the kernel Realm. So I think this enables a sandbox breach.

Maybe evaluate a short function that closes over the real one?

```js
const reg2 = s.evaluate(`f => registerEndOfCrank(f)`, { registerEndOfCrank });
```

This isn't very robust: if `registerEndOfCrank` throws an exception, the static vat can use the exception object to breach the Realm. In other places we carefully catch exceptions in one realm and wrap them with objects from the other. This will all get a lot easier in new-SES, but for now we should at least make it clear that there's a realm boundary going on.

Reply:
Done.